rmc/utils/mpc.py

```diff
@@ -336,35 +385,55 @@ def prepare_pop_path_ht(
     ht = ht.annotate(cadd=hl.struct(**cadd_ht[ht.key]))
 
     logger.info("Getting transcript annotations...")
-    # Create VEP context HT filtered to missense variants in canonical transcripts
-    # if they don't already exist
-    if not file_exists(context_with_oe_dedup.path):
-        create_context_with_oe(overwrite=overwrite)
+    # If OE-annotated VEP context HT filtered to missense variants
+    # in canonical transcripts does not exist, create it
+    # If HT exists but without specified version of OE annotation, add that OE annotation
+    transcript_col = "transcript_rescue" if include_rescue else "transcript"
+    if not file_exists(context_with_oe_dedup.path) or (
+        transcript_col not in set(context_with_oe_dedup.ht().row)
+    ):
+        create_context_with_oe(include_rescue=include_rescue, overwrite=overwrite)
 
     # Get transcript annotation from deduplicated context HT resource
     context_ht = context_with_oe_dedup.ht()
-    ht = ht.annotate(transcript=context_ht[ht.key].transcript)
+    ht = ht.annotate(transcript=context_ht[ht.key][transcript_col])
     ht = ht.checkpoint(
-        f"{TEMP_PATH_WITH_DEL}/joint_clinvar_gnomad_transcript.ht",
+        f"{TEMP_PATH_WITH_FAST_DEL}/joint_clinvar_gnomad_transcript.ht",
         _read_if_exists=not overwrite,
         overwrite=overwrite,
     )
 
     logger.info(
         "Getting PolyPhen-2, codon, and regional missense constraint missense o/e annotations..."
     )
-    # Get Polyphen, codon, o/e, amino acid annotations
     context_ht = context_with_oe.ht()
-    context_ht = context_ht.key_by("locus", "alleles", "transcript")
+    # Re-adjust columns of context HT based on whether rescue RMC results are included in calculations
+    # If including rescue RMC results in calculations, drop initial-only OE column
+    # If not including, drop rescue-included OE column
+    context_ht_cols = set(context_ht.row)
+    if include_rescue:
+        context_ht = context_ht.transmute(oe=context_ht.oe_rescue)
+    elif "oe_rescue" in context_ht_cols:
+        context_ht = context_ht.drop("oe_rescue")
+    # Get Polyphen, codon, o/e, amino acid annotations
     ht = ht.annotate(**context_ht[ht.locus, ht.alleles, ht.transcript])
     ht = ht.checkpoint(
-        f"{TEMP_PATH_WITH_DEL}/joint_clinvar_gnomad_transcript_aa.ht",
+        f"{TEMP_PATH_WITH_FAST_DEL}/joint_clinvar_gnomad_transcript_aa.ht",
         _read_if_exists=not overwrite,
         overwrite=overwrite,
     )
 
     logger.info("Getting missense badness annotation...")
     mb_ht = misbad.ht()
+    # Re-adjust columns of misbad HT based on whether rescue RMC results are included in calculations
+    # If including rescue RMC results in calculations, drop initial-only misbad column
+    # If not including, drop rescue-included misbad column
+    mb_ht_cols = set(mb_ht.row)
+    if include_rescue:
```

Comment:
it just occurred to me, should we add a check here to see if the rescue column exists (and error out if not)?